nmap.go

```diff
@@ -269,6 +279,72 @@ func (s *Scanner) RunWithProgress(liveProgress chan<- float32) (result *Run, war
 	}
 }
 
+// RunWithStreamer runs nmap synchronously. The xml output will be written directly to a file.
+// It uses a streamer interface to constantly stream the stdout.
+func (s *Scanner) RunWithStreamer(stream Streamer, file string) (warnings []string, err error) {
```

Comment:
WDYT of removing the error here, since this func doesn't need error handling?

```suggestion
func (s *Scanner) RunWithStreamer(stream Streamer, file string) (warnings []string) {
```

Reply:
I think that is a good question. The error will show that the starting process could not launched. The warnings instead are stdout errors that do not mean a failed run.

- Instead of writing crucial errors to warnings I could return an error (l. 320-325)

WDYT of the proposed change?

Reply:
Sounds good to me

Reply:
I changed these lines :)